doc/source/users_guide/cime-customize.rst

```diff
@@ -0,0 +1,84 @@
+.. _customizing-cime:
+
+===========================
+CIME config and hooks
+===========================
+
+CIME provides the ability to define model specific config and hooks.
+
+The config alters CIME's runtime and the hooks are triggered during their event.
+
+-----------------------------------
+How does CIME load customizations?
+-----------------------------------
+
+CIME will search ``cime_config/customize`` and load any python found under this directory or it's children.
+
+Any variables, functions or classes loaded are available from the ``CIME.customize`` module.
+
+---------------------------
+CIME config
+---------------------------
+
+Available config and descriptions.
+
+=================================  ================  ====  =============================================================================================================================================================
+Variable                           Default           Type  Description                                                                                                                                                  
+=================================  ================  ====  =============================================================================================================================================================
+allow_unsupported                  True              bool  Allows creation of case that is not test or supported.
+archive_dart_component             True              bool  Archives dart component.
+archive_drv_component              True              bool  Archives drv component.
+baseline_store_teststatus          True              bool  Upon the completion of a SystemTest and GENERATE_BASELINE is True, the TestStatus will be copied from the case directory to the baseline directory.
+build_cime_component_lib           True              bool  Build CIME component lib.
+build_model_use_cmake              False             bool  When building model use CMake.
+calculate_mode_build_cost          False             bool  Calculates model build cost rather than using static value in test_scheduler
+case_setup_generate_namelist       False             bool  Creates namelist during case.setup for some tests.
+cesm_create_test_flags             True              bool  Enables CESM specific `create_test` flags.
```

Comment:
@jedwards4b I have been trying to think of a better solution for this, I'd like to avoid anything resembling `get_model() == '...'` but I'm not sure there's a better option.